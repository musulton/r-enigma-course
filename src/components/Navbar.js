import React from "react";
import {Button, Container, Navbar, Nav} from 'react-bootstrap'
import {NavLink} from "react-router-dom";
import styled from "styled-components";
import constants from "../constants";

const StyledNav = styled(Nav)`
  flex-direction: row;
`

const menu = [
    { path: constants.routes.COURSE_LIST, menuName: "Courses" },
    { path: "#", menuName: "Course Types" },
]

const NavBar = () => {

    return (
        <Navbar bg="light" expand="light" sticky={"top"}>
            <Container>
                <Navbar.Brand>Enigma Course</Navbar.Brand>
                <StyledNav>
                    {menu?.map((item, index) => (
                        <NavLink
                            to={item.path}
                            className="nav-link mx-3"
                            key={index}
                        >
                            {item.menuName}
                        </NavLink>
                    ))}
                    <Button variant="danger" className="mx-3">
                        Logout
                    </Button>
                </StyledNav>
            </Container>
        </Navbar>
    )
}

export default NavBar;