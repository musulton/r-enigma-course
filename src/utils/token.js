const TOKEN = "token";

export const getToken = () => {
    const tokenString = localStorage.getItem(TOKEN);
    return JSON.parse(tokenString);
};

export const setToken = token => {
    localStorage.setItem(TOKEN, JSON.stringify(token));
};

export const removeToken = () => {
    localStorage.removeItem(TOKEN);
}
