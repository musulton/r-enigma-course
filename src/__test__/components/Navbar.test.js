import Navbar from "../../components/Navbar";
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { removeToken } from "../../utils/token";
import { BrowserRouter } from "react-router-dom";

jest.mock("../../utils/token", () => ({
  removeToken: jest.fn()
}))

describe("Navbar component", () => {
  test("should render correctly", () => {
    const {asFragment} = render(<Navbar />, {wrapper: BrowserRouter})

    expect(asFragment()).toMatchSnapshot()
  })

  test("should call remove token when onLogout is clicked", () => {
    render(<Navbar />, {wrapper: BrowserRouter})

    const logoutBtn = screen.getByText("Logout")
    userEvent.click(logoutBtn)

    expect(removeToken).toBeCalled()
  })
})
