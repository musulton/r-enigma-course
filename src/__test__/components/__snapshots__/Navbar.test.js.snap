// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-light navbar-light bg-light sticky-top"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
      >
        Enigma Course
      </span>
      <div
        class="sc-bgqQcB koDmhZ navbar-nav"
      >
        <a
          class="nav-link mx-3"
          href="/course-list"
        >
          Courses
        </a>
        <a
          aria-current="page"
          class="nav-link mx-3 active"
          href="/"
        >
          Course Types
        </a>
        <button
          class="mx-3 btn btn-danger"
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
